sdk/python/kfp/components/executor.py

```diff
@@ -41,13 +41,16 @@ def __init__(self, executor_input: Dict, function_to_execute: Callable):
         self._output_artifacts[name] = self._make_output_artifact(
             artifacts_list[0])
 
+    self._return_annotation = inspect.signature(self._func).return_annotation
+    self._executor_output = {}
+
   @classmethod
-  def _make_input_artifact(self, runtime_artifact: Dict):
+  def _make_input_artifact(cls, runtime_artifact: Dict):
```

Comment:
Should this be a static method or maybe just a normal function?

Reply:
No. Classmethods let me easily include it in lightweight components by source. Static methods are discouraged: https://google.github.io/styleguide/pyguide.html#217-function-and-method-decorators